packages/mobile/src/web3/dataEncryptionKey.ts

```diff
@@ -103,10 +126,53 @@ export function* registerAccountDek(account: string) {
     }
 
     // Generate and send a transaction to set the DEK on-chain.
-    const setAccountTx = accountsWrapper.setAccount('', publicDataKey, account)
+    let setAccountTx = accountsWrapper.setAccount('', publicDataKey, walletAddress)
     const context = newTransactionContext(TAG, 'Set wallet address & DEK')
-    yield call(sendTransaction, setAccountTx.txo, account, context)
+    // If MTW has been created, route the user's DEK/wallet registration through it
+    // because accountAddress is determined by msg.sender
+    if (mtwAddress) {
+      try {
+        const mtwWrapper: MetaTransactionWalletWrapper = yield call(
+          [contractKit.contracts, contractKit.contracts.getMetaTransactionWallet],
+          mtwAddress
+        )
+
+        const proofOfPossession: {
+          v: number
+          r: string
+          s: string
+        } = yield call(accountsWrapper.generateProofOfKeyPossession, mtwAddress, walletAddress)
+
+        // TODO: There should be a branching off point here depending on if there
+        // is an active Komenci session (i.e., submit through KomenciKit or not)
+
+        setAccountTx = accountsWrapper.setAccount(
+          '',
+          publicDataKey,
+          walletAddress,
+          proofOfPossession
+        )
+
+        const setAccountTxViaMTW: CeloTransactionObject<string> = yield call(
+          mtwWrapper.signAndExecuteMetaTransaction,
+          setAccountTx.txo
+        )
+        yield call(sendTransaction, setAccountTxViaMTW.txo, walletAddress, context)
+        yield put(updateWalletToAccountAddress({ [walletAddress.toLowerCase()]: mtwAddress }))
+      } catch (error) {
+        Logger.debug(
+          `${TAG}@registerAccountDek`,
+          'Failed to register via the MTW, will attempt with EOA. Error: ',
```

Comment:
I don't think this is the right fallback behavior?

Reply:
You're right, not what we want to do here.